### DISCUSSION TITLE
Show Virtual Chasis Instead of Individual Members

### DISCUSSION THREAD

**Discussion by adparis99** [mattieserver/netbox-topology-views#218] [Ideas]
I thought about making this a feature request, but without any knowledge of how much work this would take I figured it might be better to start here. Would it be possible to have an option to display virtual chasis as opposed to individual switch members? For example, if I have a virtual chasis called IDF_A, with member devices IDF_A_01-3, three devices are shown on the topology. I know this is a more accurate representation of the physical topology, but from a logical perspective it can be confusing at a glance. 

I imagine it would be a checkbox on the filter page that checks if a device is part of a VC, and if it is, if it is the master. If not, it would be hidden from view. I know the ORM supports the ability to get all member interfaces via the vc_interfaces() method; would it be possible to then get connections from there? This is all me spitballing, so please feel free to provide feedback and corrections if this isn't feasible.

**Comment by dreng** [comment]
This should hardly be possible because Virtual Chassis hasn't got any ports or interfaces. NTV shows devices and the connections between their ports and interfaces.

**Reply by adparis99** [reply]
I totally get that - maybe I could've worded my thoughts better. If a group of devices belong to a virtual chassis, I'd like: 
- only one device (probably the master) of the chassis to be shown on the topology
- for the device to display the name of the chassis, not itself
- for the device to show all interfaces/connections belonging to the chassis, not just itself

Hopefully that makes more sense. Some other thoughts:
- I originally imagined it could be possible because of the device.vc_interfaces() method that can grab all interfaces of a virtual chassis that a device is associated with
- I'm not sure if that would also allow for getting connections of said interfaces
- I'm not sure if this goes against the idea of the plugin; I know it's been a mostly physical representation to this point, but with L3 topologies being worked on I thought this would be a helpful and logical addition

Again, not sure how on or off-base I am, feel free to correct me.
